### PR TITLE
feat: add aria suite lifecycle functions

### PR DESCRIPTION
**Pull request by burnsjared0415**
### Summary

Added two new functions for Aria Suite Lifecycle:

- `Get-vRSLCMEnvironmentVMs`
- `Get-vRSLCMProductDetails`

### Type

- [ ] Bugfix
- [x] Enhancement or Feature
- [ ] Code Style or Formatting
- [ ] Documentation
- [ ] Refactoring
- [ ] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### Test and Documentation

- [x] Tests have been completed.
- [x] Documentation has been added or updated.

### CODE REVIEW

File: PowerValidatedSolutions.psm1 [hunk 1]
```diff
@@ -37460,6 +37460,101 @@ Function Start-vRSLCMUpgrade {
...
+        $response
+        } else {
+        Write-PowerManagementLogMessage -Type INFO -Message "$productid is not installed in Aria Lifecycle Manager" -Colour Yellow
+        }
+        
+    } Catch {
+        Write-Error $_.Exception.Message
+    }
+} 
+Export-ModuleMember -Function Get-vRSLCMProductDetails
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:37502]: Some formatting...

```suggestion
Function Get-vRSLCMProductDetails {
    <#
        .SYNOPSIS
        Get product details in VMware Aria Suite Lifecycle.

        .DESCRIPTION
        The Get-vRSLCMProductDetails cmdlet gets the product details for a certain product in VMware Aria Suite Lifecycle.

        .EXAMPLE
        Get-vRSLCMProductDetails -productid vrli
        This example gets all environments in VMware Aria Suite Lifecycle for VMware Aria Operations for Logs.

        .PARAMETER productid
        The product ID of the product to get the details for.
    #>

    Param (
        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$productid
    )

    Try {
        $productname = Get-vRSLCMEnvironment | where-object{$_.products.id -eq $productid }
        if ($productname -ne $null) {
            $envid = $productname.environmentId
            $uri = "https://$vrslcmAppliance/lcm/lcops/api/environments/$envid/products/$productid"
            $response = Invoke-RestMethod $uri -Method 'GET' -Headers $vrslcmHeaders
            $response
        } else {
            Write-Warning -Type INFO -Message "$productid is not installed in VMware Aria Suite Lifecycle: PRE_VALIDATION_FAILED"
        }   
    } Catch {
        Write-Error $_.Exception.Message
    }
}
Export-ModuleMember -Function Get-vRSLCMProductDetails
```

File: PowerValidatedSolutions.psm1 [hunk 2]
```diff
@@ -37460,6 +37460,101 @@ Function Start-vRSLCMUpgrade {
...
+                }
+            }
+        }
+        return $nodelist
+    } Catch {
+        Write-Error $_.Exception.Message
+    }
+
+} 
+Export-ModuleMember -Function Get-vRSLCMEnvironmentVMs
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:37555]: Formatting and addition of an else:

```suggestion
Function Get-vRSLCMEnvironmentVMs {
    <#
        .SYNOPSIS
        Get product virtual vachines in VMware Aria Suite Lifecycle.

        .DESCRIPTION
        The Get-vRSLCMEnvironmentVMs cmdlet gets the virtual machines that exist for certain product in VMware Aria Suite Lifecycle.

        .EXAMPLE
        Get-vRSLCMEnvironmentVMs -server sfo-sddc01.rainpole.io -user administrator -pass VMw@re1! -productid vrli
        This example gets virtual machines that exist for Aria Operations for Logs that existing in VMware Aria Suite Lifecycle.

        .PARAMETER server
        The fully qualified domain name of the SDDC Manager.

        .PARAMETER user
        The username to authenticate to the SDDC Manager.

        .PARAMETER pass
        The password to authenticate to the SDDC Manager.

        .PARAMETER productid
        The product ID of the product to get the details for.
    #>

    Param (
        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String]$server,
        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String]$user,
        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String]$pass,
        [Parameter (Mandatory = $false)] [ValidateSet("vidm","vra","vrli","vrni","vrops","vro","vssc")][ValidateNotNullOrEmpty()] [String]$productid
    )

    Try {
        if (Test-VCFConnection -server $server) {
            if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
                if (($vcfVrslcmDetails = Get-vRSLCMServerDetail -fqdn $server -username $user -password $pass)) {
                    if (Test-vRSLCMAuthentication -server $vcfVrslcmDetails.fqdn -user $vcfVrslcmDetails.adminUser -pass $vcfVrslcmDetails.adminPass) {
                        $nodes = (Get-vRSLCMProductDetails -productid $productid).nodes.properties.vmname
                        $nodelist = $nodes | Select-Object -Unique   
                    } else {
                        Write-Warning "Unable to authenticate to VMware Aria Suite Lifecycle $($vcfVrslcmDetails.fqdn): PRE_VALIDATION_FAILED"
                    }
                }
            }
        }
        return $nodelist
    } Catch {
        Write-Error $_.Exception.Message
    }
} 
Export-ModuleMember -Function Get-vRSLCMEnvironmentVMs
```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Get-vRSLCMEnvironmentVMs
+
+## SYNOPSIS
+
+Get Product Virtual Machines in VMware Aria Suite Lifecycle.
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md:5]:
```suggestion
Get virtual machines in VMware Aria Suite Lifecycle for a specified product.
```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+## SYNTAX
+
+```powershell
+Get-vRSLCMEnvironmentVMs [[-server] <String>] [[-user] <String>] [[-pass] <String>] [[-productid] <String>]
+ [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+The `Get-vRSLCMEnvironmentVMs`cmdlet gets the Virtual Machines that exist for certain product in VMware Aria Suite Lifecycle.
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md:16]:
```suggestion
The `Get-vRSLCMEnvironmentVMs`cmdlet gets the virtual machines that exist for certain product in VMware Aria Suite Lifecycle.
```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+## EXAMPLES
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md:18]:
```suggestion
## Examples

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
+# Get-vRSLCMEnvironmentVMs
+
+## SYNOPSIS
+
+Get Product Virtual Machines in VMware Aria Suite Lifecycle.
+
+## SYNTAX
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md:7]:
```suggestion
## Syntax

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+Get Product Virtual Machines in VMware Aria Suite Lifecycle.
+
+## SYNTAX
+
+```powershell
+Get-vRSLCMEnvironmentVMs [[-server] <String>] [[-user] <String>] [[-pass] <String>] [[-productid] <String>]
+ [<CommonParameters>]
+```
+
+## DESCRIPTION
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md:14]:
```suggestion
## Description

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+The `Get-vRSLCMEnvironmentVMs`cmdlet gets the Virtual Machines that exist for certain product in VMware Aria Suite Lifecycle.
+
+## EXAMPLES
+
+### EXAMPLE 1
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md:22]:
```suggestion
### Example 1

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+```powershell
+Get-vRSLCMEnvironmentVMs -server sfo-sddc01.rainpole.io -user administrator -pass VMw@re1! -productid vrli
+```
+
+This example gets Virtual machines that exist for Aria Operations for Logs that existing in VMware Aria Suite Lifecycle.
+
+## PARAMETERS
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md:31]:
```suggestion
## Parameters

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+Required: False
+Position: 1
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -user
+
+The username to connect to the SDDC Manager.
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md:51]:
```suggestion
The username to authenticate to the SDDC Manager.

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+Required: False
+Position: 2
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -pass
+
+The password to connect to the SDDC Manager.
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md:67]:
```suggestion
The password to authenticate to the SDDC Manager.

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+Aliases:
+
+Required: False
+Position: 4
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### CommonParameters
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md:97]:
```suggestion
### Common Parameters

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+
+This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMEnvironmentVMs.md:99]:
```suggestion
This cmdlet supports the common parameters: `-Debug`, `-ErrorAction`, `-ErrorVariable`, `-InformationAction`, -`InformationVariable`, `-OutVariable`, `-OutBuffer`, `-PipelineVariable`, `-Verbose`, `-WarningAction`, and `-WarningVariable`. For more information, see [about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+Aliases:
+
+Required: True
+Position: 1
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### CommonParameters
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md:45]:
```suggestion
### Common Parameters

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+
+This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md:47]:
```suggestion
```suggestion
This cmdlet supports the common parameters: `-Debug`, `-ErrorAction`, `-ErrorVariable`, `-InformationAction`, -`InformationVariable`, `-OutVariable`, `-OutBuffer`, `-PipelineVariable`, `-Verbose`, `-WarningAction`, and `-WarningVariable`. For more information, see [about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).

```
```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+
+```powershell
+Get-vRSLCMProductDetails [-productid] <String> [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+The `Get-vRSLCMProductDetails` cmdlet gets the product details for a certain product in VMware Aria Suite Lifecycle.
+
+## EXAMPLES
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md:17]:
```suggestion
## Examples

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+### EXAMPLE 1
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md:19]:
```suggestion
### Example 1

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md [hunk 18]
```diff
@@ -0,0 +1,47 @@
+# Get-vRSLCMProductDetails
+
+## SYNOPSIS
+
+Get Product Details in VMware Aria Suite Lifecycle.
+
+## SYNTAX
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md:7]:
```suggestion
## Syntax

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md [hunk 19]
```diff
@@ -0,0 +1,47 @@
+# Get-vRSLCMProductDetails
+
+## SYNOPSIS
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md:3]:
```suggestion
## Synopsis

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md [hunk 20]
```diff
@@ -0,0 +1,47 @@
+# Get-vRSLCMProductDetails
+
+## SYNOPSIS
+
+Get Product Details in VMware Aria Suite Lifecycle.
```
Comment (tenthirtyam) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductDetails.md:5]:
```suggestion
Get the product details in VMware Aria Suite Lifecycle.

```

File: PowerValidatedSolutions.psm1
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:1]: What if you have >1 deployment in Aria Suite Lifecycle for a product? There's not an environment filter.

Example - one in VCF-mode and one not.

Comment (burnsjared0415) [reply PowerValidatedSolutions.psm1:1]: i will take a look, to see how you would handle multiple

Comment (burnsjared0415) [reply PowerValidatedSolutions.psm1:1]: add new code to handle multiple deployments

**Review by tenthirtyam — APPROVED** [review]
LGTM! 🎉

**Pull request merged**